ParabolInc/parabol
# fix(Jira): show only available fields in Sprint Poker

# Description

Partially Fixes #7206

Part 1/3
* store the estimation field per issue type and in a separate table than team (aligned with other integrations)
* if no estimation field is stored for the specific issue type, fall back to fields from other issue types from the same project (if these match). This also allows transitioning from the current state were no issue types are stored
* available fields are read from `editmeta` instead of `schema` which supposedly only shows fields that can be edited. This change will hide most fields from the menu.

## Demo

### new JiraDimensionFieldMap table
https://www.loom.com/share/c87e4bc3fa2740e199885401c7cf429e

### reading and voting to different fields
https://www.loom.com/share/69526435b3504b98b02c7b91242cfdd6

## Testing scenarios

- See demos

- [ ] Update a field after migration
  - check Postgres JiraDimensionFieldMap table before the change, it should contain an entry for the `{teamId, cloudId, projectKey}` tuple with `issueType===''`
  - start a sprint poker with and estimate a matching Jira issue
  - [ ] check DB again and now see the correct `issueType`
  - estimate another issue type
  - [ ] see new entry in DB

- [ ] Fallback field is used correctly
  - setup a Jira project with 2 issue types, both with the same field "Story Points"
  - estimate one issue in a poker meeting and set the field to "Story Points"
  - start a new poker meeting and add the second issue type
  - [ ] the selected field should also be "Story Points"
  - [ ] after voting on this issue, a second entry is added to the DB

- [ ] Different voting fields for different issue types
  - set up a Jira project with 2 issue types, both have "Story Points" but one has an additional field "Exclusive Task Points"
  - run a poker meeting and choose different fields to estimate for both
  - [ ] run a second meeting and see that different fields are stored for different issue types

## Final checklist
